File: easy/3010/main.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
       int min = nums[0];
       vector<int> a(2, INT_MAX);
       for (int i = 1; i < nums.size(); i++) {
         if (nums[i] < a[0]) { 
            a[1] = a[0];
            a[0] = nums[i];
         } else if (nums[i] < a[1]) {
            a[1] = nums[i];
         }
       }
        
       return min + a[0] + a[1];
    }
};

File: easy/3010/main.rs
impl Solution {
    pub fn minimum_cost(nums: Vec<i32>) -> i32 {
        let mut min = nums[0];
        let mut a = vec![i32::MAX, i32::MAX];
        for num in &nums[1..] {
            if *num < a[0] {
                a[1] = a[0];
                a[0] = *num;
            } else if *num < a[1] {
                 a[1] = *num;
            }
        }

        min + a[0] + a[1]
        
    }
}
